### ISSUE TITLE
Wow.. thanks a lot! :-) &CLI?

### ISSUE THREAD

**Issue by hoijui** [atextor/turtle-formatter#1]
So cool!! :D
I have not tried it yet, but I will today.
I want to use it as a CLI tool, to use in a git pre-commit hook.
What do you think... would you want a CLI interface in the project, or should it rather be an other project, using this library?

and yeah.. thanks a lot for it!
made me happy finding it!. :-)

**Comment by atextor** [comment]
Hello @hoijui,
thank you, your comment made my day :)
Yes, having CLI for the turtle-formatter is a good idea, and incidentally, a CLI already exists! turtle-formatter was originally conceived as a satellite/supporting project for the [owl-cli](https://github.com/atextor/owl-cli), an allround RDF/OWL command line tool which can, among other thinks, pretty-print RDF/Turtle documents. So you can [get the latest release](https://github.com/atextor/owl-cli/releases/tag/v1.1.0) (.exe or MacOS/Linux binary), rename it to `owl` or `owl.exe` and run `owl write mydocument.ttl` to pretty-print to stdout (it does work with both OWL and plain RDF). In the [documentation](https://atextor.de/owl-cli/main/snapshot/usage.html#write-command) you can find the command line switches to adjust the formatting.

However, what is obviously missing here is a corresponding hint in the README. I'll add this soon. Let me know if this helps.

**Comment by atextor** [comment]
Yes, JVM bootup is a bit slow, which is why I evaluated a bunch of possibilities to remedy that and I wrote a post on that: https://atextor.de/2020/07/27/building-a-decent-java-cli.html
tl;dr: It's done with [GraalVM](https://www.graalvm.org/), which can compile Java to native binaries (with a bit of work on the author's side to deal with resources, reflection, JNI etc.).
If you'd like to build the owl-cli native binary yourself, that's documented [here](https://atextor.de/owl-cli/main/snapshot/index.html#building-native-image).
